llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/metrics/LlapMetricsCollector.java

```diff
@@ -58,26 +61,44 @@
   private final Map<String, LlapManagementProtocolClientImpl> llapClients;
   private final Map<String, LlapMetrics> instanceStatisticsMap;
   private final long metricsCollectionMs;
+  @VisibleForTesting
+  final LlapMetricsListener listener;
```

Comment:
Would it potentially make sense to have multiple listeners?

Reply:
I was thinking about the same lines, but we do not need them now, so I decided to keep this as simple as possible